US-Trustee-Program/Bankruptcy-Oversight-Support-Systems
# CAMS-121 Update frontend host instance

# Purpose

Switch from using a Windows host to Linux host for Application Service.

# Major Changes

- Update Azure App Service plan to use Linux base host
- Add flag `--os-type` for az deployment command
- Replace web.config with nginx.conf
- Update base nginx.conf to fix page routes and remove references to php

# Testing/Validation

- Provision and deploy branch changes successfully. Manually verified ui loads into web browser and navigation to existing views.
- Resolve issue found for URI routing. To test, the following steps can be taken:
  - Navigate to CAMS home page
  - Navigate to Cases or Assignment view and let it load
  - Reload page or navigating directly to the uri of the new view should not return a 404

# Notes

- For deploying to existing deployment, some manual clean up of Azure resource will be needed. The Azure app service and app service plan will need to be deleted before deploying.
- Web server on the host instance has changed from IIS to Nginx. Added nginx.conf to user-interface/public folder to include the new server configs.
- Note that php runtime was used based on Azure App Service automated selection of a container to host the web application project. A container with NodeJs runtime could be configured but that would require additional setup to include a web server since this container will not include nginx.
